src/definitions.rs

```diff
@@ -43,69 +42,7 @@ pub enum Action {
     Put = 13,
     Unhook = 14,
     BeginUtf8 = 15,
-}
-
-/// Unpack a u8 into a State and Action
-///
-/// The implementation of this assumes that there are *precisely* 16 variants for both Action and
-/// State. Furthermore, it assumes that the enums are tag-only; that is, there is no data in any
-/// variant.
-///
-/// Bad things will happen if those invariants are violated.
-#[inline(always)]
-pub fn unpack(delta: u8) -> (State, Action) {
-    unsafe {
-        (
-            // State is stored in bottom 4 bits
-            mem::transmute(delta & 0x0f),
-            // Action is stored in top 4 bits
-            mem::transmute(delta >> 4),
-        )
-    }
-}
```

Comment:
This needs to be benchmarked. There is no way we'll merge a patch that reduces performance for a feature nobody should be using.

Reply:
Running some initial benchmarks I've got the following results from `cargo bench`:

```
test bench::state_changes ... bench:     108,197 ns/iter (+/- 3,199)
test bench::testfile      ... bench:     121,021 ns/iter (+/- 3,617)
```

whilst the old code yields the following:

```
test bench::state_changes ... bench:      95,320 ns/iter (+/- 4,951)
test bench::testfile      ... bench:     101,009 ns/iter (+/- 7,342)
```

Which equates to roughly an 12% ~ 16% decrease in performance.  I am using flamegraph now to investigate.

Reply:
I'd also recommend testing with https://github.com/alacritty/vtebench. That's ultimately what decides which features get merged into Alacritty or not.